layer1/executor/tasks/dkg/mpk_submission.go

```
@@ -125,7 +126,22 @@ func (t *MPKSubmissionTask) Execute(ctx context.Context) (*types.Transaction, *t
 
 	// submit if I'm a leader for this task
 	client := t.GetClient()
-	if !utils.AmILeading(client, ctx, logger, int(t.GetStart()), t.StartBlockHash.Bytes(), dkgState.NumberOfValidators, dkgState.Index) {
+	isLeading, err := utils.AmILeading(
+		client,
+		ctx,
+		logger,
+		int(t.GetStart()),
+		t.StartBlockHash.Bytes(),
+		dkgState.NumberOfValidators,
+		dkgState.Index-1,
+		constants.ETHDKGDesperationFactor,
+		constants.ETHDKGDesperationDelay,
+	)
+	if err != nil {
+		return nil, tasks.NewTaskErr("error getting eth height for leader election", true)
```

`err` gets swallowed. Maybe log it before returning a different error?